src/Expression.php

```diff
@@ -528,9 +529,13 @@ public function execute(object $connection = null): object
                     if (is_int($val)) {
                         $type = \PDO::PARAM_INT;
                     } elseif (is_bool($val)) {
-                        // SQL does not like booleans at all, so convert them INT
-                        $type = \PDO::PARAM_INT;
-                        $val = (int) $val;
+                        if ($this->connection->getDatabasePlatform() instanceof PostgreSQL94Platform) {
+                            $type = \PDO::PARAM_STR;
+                            $val = $val ? '1' : '0';
+                        } else {
+                            $type = \PDO::PARAM_INT;
```

Comment:
At least Oracle does not support `PARAM_BOOL` with `col = false` in where condition.